File: WoofAndroid/data/datasource/src/testFixtures/kotlin/com/perrystreet/woof/datasource/account/AccountDataSourceFactory.kt
package com.perrystreet.woof.datasource.account

import org.koin.test.KoinTest
import org.koin.test.inject

class AccountDataSourceFactory : KoinTest {
    private val dataSource: FakeAccountDataSource by inject()

    fun withAccountError() = apply {
        dataSource.getAccountError = IllegalStateException("Could not load account")
    }
}

File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSourceFakes/Account/AccountDataSourceFactory.swift
import DataSource
import Swinject
import SwinjectAutoregistration

public final class AccountDataSourceFactory {
    private let dataSource: FakeAccountDataSource

    public init(_ container: Container) {
        dataSource = container~>
    }

    @discardableResult
    public func withAccountError() -> Self {
        dataSource.getAccountError = .unavailable("Could not load account")
        return self
    }
}
